src/amo/components/SearchForm.js

```diff
@@ -25,8 +25,9 @@ export class SearchFormBase extends React.Component {
 
   goToSearch(query) {
     const { api, pathname } = this.props;
+    const encodedQuery = query ? encodeURIComponent(query) : '';
     this.context.router.push(
```

Comment:
Oh, true. I guess the previous way was just too naive (with the string interpolation). 👍